"Refactor player's angle calculation for game updates

BenoitRoux0 opened the pull request:
This commit simplifies angle calculations for player movements and extends the functionality by extracting it to a new struct, t_angle_data. Instead of calculating radian, cosine, and sine each time, they are now part of player's struct and calculated when the player's orientation changes. This update reduces repetitive computation, simplifies the code and potentially reducing runtime, making the application more efficient."

BenoitRoux0 committed:
"Refactor player's angle calculation for game updates

This commit simplifies angle calculations for player movements and extends the functionality by extracting it to a new struct, t_angle_data. Instead of calculating radian, cosine, and sine each time, they are now part of player's struct and calculated when the player's orientation changes. This update reduces repetitive computation, simplifies the code and potentially reducing runtime, making the application more efficient."

BenoitRoux0 commented:
Effet de bord de l'opti: tout le rendu est plus rapide

gd-harco commented on bonus/incs/types_bonus.h:
```diff
@@ -121,7 +129,7 @@ typedef struct s_data
 	t_player		player;
 	t_ray			rays[WIN_WIDTH];
-	int 			show_minimap;
+	int				show_minimap;
```
Je maintien la requête d'utiliser un booléen dans ce genre de cas pour faciliter la lisibilité du code

gd-harco reviewed (commented):
I'd rather use Boolean for the "show minimal" check in data.

BenoitRoux0 committed:
Add init_angles.c and refactor angle calculations

Implement angle calculations with the introduction of t_angle_data struct, associating cosine and sine values with the player structure, resulting in minimized computations and simplified code structure. Division operations are replaced by bitwise right shift operations for performance optimization. Update file headers and make code formatting adjustments where necessary.

BenoitRoux0 committed:
Update Makefile to include `init_angles.c`

This commit introduces a new file, `init_angles.c`, into the Makefile for the build of the bonus version of cub3D. This was done to improve angle calculations by creating a new struct, `t_angle_data`, to store cosine and sine values associated with the player. This change allows minimization of computations and a simplification of the code structure. It also improves performance by replacing division operations with bitwise right shift operations. Additionally, file headers are updated and some code formatting adjustments are made where necessary.

BenoitRoux0 committed:
Fix texture mapping issue and improve performance

The modification in `render.c` and `render_bonus.c` fixes incorrect texture mapping, caused by wrong calculation of src_x. Consequently, the calculation of 'src_x' has been revised. A division operation `% tex->width` has been replaced by a multiplication and division operation `* tex->width / CELL_SIZE`. This accurate calculation of 'src_x' ensures correct texture display.

The change in 'draw_bonus.c' replaces division by 8 (to compute number of bytes per pixel) with a bitwise right shift operation `>> 3` which serves the same purpose. This change not only simplifies the expression but also improves code efficiency, as bitwise operations are faster than arithmetic operations.

BenoitRoux0 committed:
Change type of 'show_minimap' variable to bool

The 'show_minimap' variable in 'types_bonus.h' has been changed from an integer to a bool. This change helping to express the semantic purpose of the variable more clearly, which is just to show or hide the minimap.
